packages/thirdweb/src/react/web/wallets/ecosystem/EcosystemWalletFormUI.tsx

```diff
@@ -0,0 +1,375 @@
+"use client";
+import styled from "@emotion/styled";
+import { useState } from "react";
+import type {
+  InAppWalletAuth,
+  InAppWalletSocialAuth,
+} from "../../../../wallets/in-app/core/wallet/types.js";
+import type { Wallet } from "../../../../wallets/interfaces/wallet.js";
+import type { EcosystemWalletId } from "../../../../wallets/wallet-types.js";
+import { useCustomTheme } from "../../../core/design-system/CustomThemeProvider.js";
+import {
+  fontSize,
+  iconSize,
+  spacing,
+} from "../../../core/design-system/index.js";
+import { useConnectUI } from "../../../core/hooks/others/useWalletConnectionCtx.js";
+import { useSetSelectionData } from "../../providers/wallet-ui-states-provider.js";
+import { TOS } from "../../ui/ConnectWallet/Modal/TOS.js";
+import { useScreenContext } from "../../ui/ConnectWallet/Modal/screen.js";
+import { PoweredByThirdweb } from "../../ui/ConnectWallet/PoweredByTW.js";
+import { WalletTypeRowButton } from "../../ui/ConnectWallet/WalletTypeRowButton.js";
+import {
+  emailIcon,
+  passkeyIcon,
+  phoneIcon,
+} from "../../ui/ConnectWallet/icons/dataUris.js";
+import { Img } from "../../ui/components/Img.js";
+import { Spacer } from "../../ui/components/Spacer.js";
+import { TextDivider } from "../../ui/components/TextDivider.js";
+import { Container } from "../../ui/components/basic.js";
+import { Button } from "../../ui/components/buttons.js";
+import { InputSelectionUI } from "../in-app/InputSelectionUI.js";
+import { socialIcons } from "../in-app/socialIcons.js";
+import type { InAppWalletSelectUIState } from "../in-app/types.js";
+import { validateEmail } from "../in-app/validateEmail.js";
+import type { InAppWalletLocale } from "../shared/locale/types.js";
+import { openOauthSignInWindow } from "../shared/openOauthSignInWindow.js";
+import { setLastAuthProvider } from "../shared/storage.js";
+import { EcosystemWalletHeader } from "./EcosystemWalletHeader.js";
+
+const defaultAuthOptions: InAppWalletAuth[] = [
+  "email",
+  "phone",
+  "google",
+  "apple",
+  "facebook",
+  "passkey",
+];
+
+export type InAppWalletFormUIProps = {
+  select: () => void;
+  done: () => void;
+  locale: InAppWalletLocale;
+  wallet: Wallet<EcosystemWalletId>;
+  goBack?: () => void;
+};
+
+/**
+ * @internal
+ */
+const EcosystemWalletFormUI = (props: InAppWalletFormUIProps) => {
```

Comment:
bunch of duplicated code with InApp ui - was there no way to re-use some of this code?

Reply:
Did some more refactoring and pulled a bunch of it out to `ConnectWalletSocialOptions`, let me know if there's anywhere else you think we could reuse something (intentionally duplicated InAppWalletConnectUI because I think we're planning on adding some connection options unique to ecosystem wallets)

Reply:
looks much better!